hack/deploy/secrets.yaml

```diff
@@ -0,0 +1,44 @@
+---
+kind: Secret
+apiVersion: v1
+metadata:
+ name: mcg-osd-deadmanssnitch
+ namespace: redhat-data-federation
+stringData:
+ SNITCH_URL: https://nosnch.in/4a029adb4c
+type: Opaque
+---
+kind: Secret
+apiVersion: v1
+metadata:
+ name: mcg-osd-pagerduty
+ namespace: redhat-data-federation
+stringData:
+ PAGERDUTY_KEY: testKey
+type: Opaque
+---
+kind: Secret
+apiVersion: v1
+data:
+ host: c210cC5zZW5kZ3JpZC5uZXQ=
+ password: dGVzdC1wYXNzd29yZAo=
+ port: NTg3
+ tls: dHJ1ZQ==
+ username: YXBpa2V5
+metadata:
+ name: mcg-osd-smtp
+ namespace: redhat-data-federation
+type: Opaque
+---
+kind: Secret
+apiVersion: v1
+metadata:
+ name: addon-mcg-osd-parameters
+ namespace: redhat-data-federation
+ labels:
+   hive.openshift.io/managed: 'true'
+data:
+ notification-email-0: bmFwYXVsQHJlZGhhdC5jb20=
+ notification-email-1: bmFwYXVsQHJlZGhhdC5jb20=
+ notification-email-2: bmFwYXVsQHJlZGhhdC5jb20=
+type: Opaque
```

Comment:
Are these must-have secrets that operators require before installation?

Reply:
Yep, absence of these secrets will cause the reconciliation loop to fail.

Reply:
If these are must requirements for the operator why don't the Operator creates them as pre-requisites?

Reply:
Operator does not create them, but checks for them in the environment (these are supplied to us from the deploying entity): https://github.com/red-hat-storage/mcg-osd-deployer/blob/main/main.go#L113-L121.

Reply:
@cloudbehl this resourses are created by addon, app-interface, not by us